lnwallet/channel.go

```diff
@@ -857,14 +863,14 @@ func (lc *LightningChannel) diskCommitToMemCommit(isLocal bool,
 	var localCommitKeys, remoteCommitKeys *CommitmentKeyRing
 	if localCommitPoint != nil {
 		localCommitKeys = deriveCommitmentKeys(
```

Comment:
I don't think it needs to be a method. It doesn't critically depend on any of the attributes of `LightningChannel`. Forcing it to be a method would also mean that we're forced to create an partially populate a `LightningChannel` instance each time we want to derive or verify commitment keys. [An an example, here we create a commitment transaction during the funding phase, but the funding isn't fully completed, so we don't have a full `LightningChannel` instance yet](https://github.com/lightningnetwork/lnd/blob/master/lnwallet/wallet.go#L830).

Reply:
Yes, I know it is also used without a `LightningChannel`. That is also why I suggested to also keep a global version of the function. 

The advantage of (also) having a method is that is reduce the potential for mistakes in passing the wrong value and deduplication of code in general.

Reply:
I don't see how this change would clean up the diff, it seems marginal tbh. Also in the one case where this is now used as a public function, it wouldn't be able to use a method that say had no params (as it uses all the internal state), since it needs to possibly derive the keys for a _prior_ state (a revoked one possibly), requiring it to pass in a new custom commitment point for that state.

Reply:
I don't suggest to use all the internal state, just the chan configs and the tweakless bool. I think that a little more encapsulation could be beneficial for the maintainability of the code over time. It would also move a little bit towards abstraction of commitment formats.